source/common/http/conn_pool_grid.cc

```diff
@@ -349,11 +351,35 @@ bool ConnectivityGrid::isPoolHttp3(const ConnectionPool::Instance& pool) {
   return &pool == pools_.begin()->get();
 }
 
-bool ConnectivityGrid::isHttp3Broken() const { return http3_status_tracker_.isHttp3Broken(); }
+AlternateProtocolsCache::Origin ConnectivityGrid::getOrigin() const {
+  uint32_t port{0};
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ENVOY_LOG_MISC(error, "Address is not an IP address");
+    ASSERT(false);
+  } else {
+    port = host_->address()->ip()->port();
+  }
+  // TODO(RyanTheOptimist): Figure out how scheme gets plumbed in here.
+  return AlternateProtocolsCache::Origin("https", host_->hostname(), port);
+}
+
+bool ConnectivityGrid::isHttp3Broken() const {
+  AlternateProtocolsCache::Http3StatusTracker& http3_status_tracker =
+      alternate_protocols_->getOrCreateHttp3StatusTracker(getOrigin());
```

Comment:
Can we create a method for returning the tracker and call it from all 3 of these methods? (In fact, if we do that it looks like it will be the only caller of getOrigin() so maybe just rename getOrigin() to getTracker() and add alternate_protocols_->getOrCreateHttp3StatusTracker(...) to the end of it?